sdcm/sct_events/continuous_event.py

```diff
@@ -0,0 +1,286 @@
+
+        duration += f"{sec}s"
+
+        return duration
+
+    # TODO: rename function to "begin" after the refactor will be done
+    def begin_event(self) -> ContinuousEvent:
+        self.timestamp = time.time()
+        self.begin_timestamp = self.timestamp
+        self.period_type = EventPeriod.BEGIN.value
+        self.severity = Severity.NORMAL
+        if self.publish_event:
+            self._ready_to_publish = True
+            self.publish()
+        return self
+
+    # TODO: rename function to "end" after the refactor will be done
+    def end_event(self) -> None:
+        self.timestamp = time.time()
+        self.end_timestamp = self.timestamp
+        self.period_type = EventPeriod.END.value
+        if self.publish_event:
+            self._ready_to_publish = True
+            self.publish()
+
+    def add_error(self, errors: Optional[List[str]]) -> None:
+        if not isinstance(self.errors, list):
+            self.errors = []
+
+        self.errors.extend(errors)
+
+    # TODO: rename function to "error" after the refactor will be done
+    def event_error(self):
+        self.timestamp = time.time()
+        self.period_type = EventPeriod.INFORMATIONAL.value
+        self.duration = None
+        if self.publish_event:
+            self._ready_to_publish = True
+            self.publish()
+
+    def register_event(self):
+        self._continuous_event_registry.add_event(self)
+
+
+class ContinuousRegistryFilter:
+    def __init__(self, registry: List[Any]):
+        self._registry = registry
+        self._output = registry.copy()
+
+    def filter_by_id(self, event_id: str) -> ContinuousRegistryFilter:
+        self._output = [event for event in self._output if event.event_id == event_id]
+
+        return self
+
+    def filter_by_node(self, node: str) -> ContinuousRegistryFilter:
+        self._output = [item for item in self._output if item.node == node]
+
+        return self
+
+    def filter_by_type(self, event_type: Type[ContinuousEvent]) -> ContinuousRegistryFilter:
+        self._output = [item for item in self._output if isinstance(item, event_type)]
+
+        return self
+
+    def filter_by_period(self, period_type: EventPeriod) -> ContinuousRegistryFilter:
+        self._output = [item for item in self._output if item.period_type == period_type]
+
+        return self
+
+    def filter_by_shard(self, shard: int) -> ContinuousRegistryFilter:
+        self._output = [item for item in self._output if item.shard == shard]
+
+        return self
+
+    def filter_by_attr(self, **kwargs) -> ContinuousRegistryFilter:
+        output = []
+        for event in self._output:
+            selected = False
+            for attr, value in kwargs.items():
+                if getattr(event, attr, None) == value:
+                    selected = True
+                else:
+                    selected = False
+                    break
+
+            if selected:
+                output.append(event)
```

Comment:
Maybe even simpler?

```suggestion
        for event in self._output:
            for attr, value in kwargs.items():
                if getattr(event, attr, None) == value:
                    output.append(event)
                else:
                    break
```

Reply:
It will add it multiple times over two args

Reply:
@dkropachev is right, for every argument it will be added when match. But we want "combined" matching, it means, select the event just when ALL arguments are matched

Reply:
Ah, I see, my bad. It should stay the way it was.